lib/shellable.ts

```diff
@@ -117,14 +185,20 @@ export class Shellable extends cdk.Construct {
       buildSpec: {
         version: '0.2',
         phases: {
-          pre_build: { commands: this.platform.prebuildCommands() },
+          pre_build: { commands: this.platform.prebuildCommands(props.assumeRole) },
           build: { commands: this.platform.buildCommands(props.entrypoint) },
         }
       }
     });
 
     this.role = this.project.role;
     asset.grantRead(this.role);
+
+    if (props.assumeRole && this.project.role) {
+      this.project.role.addToPolicy(new iam.PolicyStatement()
+        .addAction('sts:AssumeRole')
+        .addResource(props.assumeRole.roleArn));
```

Comment:
Should you include the external id condition if it exists?

Reply:
The condition should be added by the destination account, not the assuming account. I don't think this would have any effect from this side.